apps/contrib/models.py

```diff
@@ -290,12 +293,65 @@ def trigger_excel_generation(self, request, model_instance_id=None):
 
 
 class Client(MetaInformationAbstractModel):
+    class USE_CASE_CHOICES(enum.Enum):
+        ANTICIPATORY_ACTION = 0
+        SITREPS = 1
+        DATA_ANALYSIS = 2
+        RESEARCH = 3
+        MODELING_DATA_SCIENCE_PROJECT = 4
+        DATA_VISUALIZATION_OR_DASHBOARD = 5
+        OTHER = 6
+
+        __labels__ = {
+            ANTICIPATORY_ACTION: _("Anticipatory action"),
+            SITREPS: _("Situation reports/analysis (SitReps)"),
+            DATA_ANALYSIS: _("Data analysis"),
+            RESEARCH: _("Research"),
+            MODELING_DATA_SCIENCE_PROJECT: _("Modeling/Data science project"),
+            DATA_VISUALIZATION_OR_DASHBOARD: _("Data visualization or dashboard"),
+            OTHER: _("Other"),
+        }
+
     name = models.CharField(max_length=255)
     code = models.CharField(
         max_length=100,
-        help_text=_('Recommended format client-short-name:custom-name:month:day'),
-        unique=True
+        unique=True,
+        editable=False  # Make it non-editable
+    )
+    acronym = models.CharField(
+        max_length=255,
+        verbose_name=_('Client Acronym'),
+        blank=True,
+        null=True
+    )
+    contact_name = models.CharField(
+        max_length=255,
+        verbose_name=_('Client Contact Name'),
+        help_text=_('Client Contact Name: focal person'),
+        blank=True,
+        null=True
     )
+    contact_email = models.EmailField(
+        verbose_name=_('Client Contact Email'),
+        help_text=_('Client Contact Email: email focal person'),
+        blank=True,
+        null=True
+    )
+    contact_website = models.URLField(
+        verbose_name=_('Client Contact Website'),
+        help_text=_('Client Contact Website: link to the website (IDMC application)'),
+        blank=True,
+        null=True
+    )
+    date_expiration = models.DateTimeField(null=True, blank=True)
+    revoked_by = models.ForeignKey(User, on_delete=models.CASCADE, null=True, blank=True,
+                                   related_name='clients_revoked')
+    revoked_at = models.DateTimeField(verbose_name=_('Revoked At'), null=True, blank=True)
+    use_case = ArrayField(
+        base_field=enum.EnumField(USE_CASE_CHOICES, verbose_name=_('Use case')),
+        blank=True, null=True
```

Comment:
Let's use this instead, 
```suggestion
        blank=True, default=list
```